dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/AbstractDataSourceProcessor.java

```diff
@@ -109,6 +109,6 @@ protected Map<String, String> transformOtherParamToMap(String other) {
     public String getDatasourceUniqueId(ConnectionParam connectionParam, DbType dbType) {
         BaseConnectionParam baseConnectionParam = (BaseConnectionParam) connectionParam;
         return MessageFormat.format("{0}@{1}@{2}@{3}", dbType.getDescp(), baseConnectionParam.getUser(),
-                PasswordUtils.encodePassword(baseConnectionParam.getPassword()), baseConnectionParam.getJdbcUrl());
+                PasswordUtils.encodePassword(baseConnectionParam.getPassword()), getJdbcUrl(connectionParam));
```

Comment:
Is it better to update `org.apache.dolphinscheduler.plugin.datasource.api.plugin.DataSourceClientProvider#uniqueId2dataSourceClientCache` while calling `updateDataSource`?

Reply:
good point

Reply:
The cache expires within 24 hours. and the cache does not necessarily need to expire when updatingDataSource. Is it better to change getDatasourceUniqueId?

Reply:
The cache expires within 24 hours, and the cache does not necessarily need to expire when updatingDataSource. Is it better to change getDatasourceUniqueId?

Reply:
Agree with @huaibingshifu's solution. Instead of updating the old datasource, I would like to create a new datasource in cache.

But it's better to generate the id by ConnectionParam's hashcode? since in some datasource there may exist other extra field.